### PR TITLE
Fix memory leak in Authentication/Safari services

### PR DESCRIPTION
**Pull request by rohitnarula7176**

### CODE REVIEW

File: IdentityCore/src/webview/systemWebview/ios/MSIDSFAuthenticationSessionHandler.m [hunk 1]
```diff
@@ -66,10 +66,12 @@ - (void)startWithCompletionHandler:(MSIDWebUICompletionHandler)completionHandler
         {
             NSError *cancelledError = MSIDCreateError(MSIDErrorDomain, MSIDErrorUserCancel, @"User cancelled the authorization session.", nil, nil, nil, nil, nil, YES);
             
+            self.webAuthSession = nil;
             if (completionHandler) completionHandler(nil, cancelledError);
             return;
         }
         
+        self.webAuthSession = nil;
         completionHandler(callbackURL, authError);
```
Comment (jasoncoolmax) [inline IdentityCore/src/webview/systemWebview/ios/MSIDSFAuthenticationSessionHandler.m]: Do we want to nil out `webAuthSession` before line 62 too?

File: IdentityCore/src/webview/systemWebview/session/MSIDASWebAuthenticationSessionHandler.m [hunk 2]
```diff
@@ -82,10 +82,12 @@ - (void)startWithCompletionHandler:(MSIDWebUICompletionHandler)completionHandler
         {
             NSError *cancelledError = MSIDCreateError(MSIDErrorDomain, MSIDErrorUserCancel, @"User cancelled the authorization session.", nil, nil, nil, nil, nil, YES);
             
+            self.webAuthSession = nil;
             if (completionHandler) completionHandler(nil, cancelledError);
             return;
         }
         
+        self.webAuthSession = nil;
```
Comment (jasoncoolmax) [inline IdentityCore/src/webview/systemWebview/session/MSIDASWebAuthenticationSessionHandler.m]: Do we want to nil out `webAuthSession` before line 76 too?

Comment (rohitnarula7176) [reply IdentityCore/src/webview/systemWebview/session/MSIDASWebAuthenticationSessionHandler.m]: Good point. We should. Addressed

File: IdentityCore/src/webview/systemWebview/ios/MSIDSFAuthenticationSessionHandler.m (hunk 1 above)
Comment (rohitnarula7176) [reply IdentityCore/src/webview/systemWebview/ios/MSIDSFAuthenticationSessionHandler.m]: We should. Addressed.

**Pull request merged**